### ISSUE TITLE
Incorrect Return Type for `S3.Client.select_object_content`

### ISSUE THREAD

**Issue by tas12740** [youtype/mypy_boto3_builder#197]
I am really enjoying this project so far and appreciate all the work that has gone into it! It's providing a lot of utility on one of the projects on which I'm working. With that said, I'm running into an error with a particular S3 call.

**Describe the bug**
As documented in the [mypy_boto3_s3 docs](https://youtype.github.io/boto3_stubs_docs/mypy_boto3_s3/type_defs/#selectobjectcontentoutputtypedef), the `Payload` key of `S3.Client.select_object_content` returns a `TypedDict`. As far as I can understand it from the [boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/select_object_content.html), the `Payload` key is a `botocore.eventstream.EventStream` ([doc](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/eventstream.html)). That is, instead of being a `dict`, it is an `Iterable` of `TypedDict` results.

**To Reproduce**

1. Install `boto3-stubs[s3]`
2. Run `mypy`/`pyright` on the following code sample

```python
from boto3 import Session

client = Session().client("s3", region_name="us-east-1")

response = client.select_object_content(
    Bucket="my-bucket",
    Key="my/key.csv",
    ExpressionType="SQL",
    # some details elided for conciseness
    Expression="my-sql-statement",
    InputSerialization={},
    OutputSerialization={},
)
for event in response["Payload"]:
    print(event["Records"]["Payload"].decode("utf-8")
```

**Actual output**

```
temp.py:15: error: "str" has no attribute "decode"; maybe "encode"?  [attr-defined]
temp.py:15: error: Invalid index type "str" for "str"; expected type "Union[SupportsIndex, slice]"  [index]
```

**Expected output**

No typing errors thrown.

**Additional context**

OS: Ubuntu 20.04 on Microsoft Windows Subsystem for Linux (WSL)
`sys.version`: `3.8.10 (default, Mar 13 2023, 10:26:41) \n[GCC 9.4.0]`
`boto3-stubs` installation method: `pip`
`boto3-stubs` version: `1.26.120`
`bogo3-stubs-s3` version: `1.26.62`
`boto3` version: `1.18.65`

**Comment by vemel** [comment]
Thank you for the report!

I will try to fix is by Monday.

**Comment by vemel** [comment]
Aaaand, finally it is Monday. :)

Adding EventStream support was more complex than I expected. Finally, the code is in place, so expect a new release tomorrow.

**Comment by vemel** [comment]
Published ` mypy_boto3_s3_package 1.28.2` with the fix included.

Please let me know if it works as it should when you have time.
